src/sundials/sundials_profiler.c

```diff
@@ -218,13 +217,11 @@ int SUNProfiler_Create(SUNComm comm, const char* title, SUNProfiler* p)
   SUNDIALS_MARK_BEGIN(profiler, SUNDIALS_ROOT_TIMER);
   sunStopTiming(profiler->overhead);
 
-  return (0);
+  return SUN_SUCCESS;
 }
 
-int SUNProfiler_Free(SUNProfiler* p)
+SUNErrCode SUNProfiler_Free(SUNProfiler* p)
 {
-  if (p == NULL) return (-1);
-
   SUNDIALS_MARK_END(*p, SUNDIALS_ROOT_TIMER);
```

Comment:
Check for `NULL` input
```suggestion
  if (!p) { return SUN_SUCCESS; }  
  
  SUNDIALS_MARK_END(*p, SUNDIALS_ROOT_TIMER);
```